libcloud/compute/drivers/ecs.py

```diff
@@ -0,0 +1,1390 @@
+        creation_time = findtext(element, 'CreationTime',
+                                 namespace=self.namespace)
+        return ECSSecurityGroup(_id, name, description=description,
+                                driver=self, vpc_id=vpc_id,
+                                creation_time=creation_time)
+
+    def _to_zone(self, element):
+        _id = findtext(element, 'ZoneId', namespace=self.namespace)
+        local_name = findtext(element, 'LocalName', namespace=self.namespace)
+        resource_types = findall(element,
+                                 'AvailableResourceCreation/ResourceTypes',
+                                 namespace=self.namespace)
+        instance_types = findall(element,
+                                 'AvailableInstanceTypes/InstanceTypes',
+                                 namespace=self.namespace)
+        disk_categories = findall(element,
+                                  'AvailableDiskCategories/DiskCategories',
+                                  namespace=self.namespace)
+
+        def _text(element):
+            return element.text
+
+        return ECSZone(id=_id, name=local_name, driver=self,
+                       available_resource_types=list(
+                           map(_text, resource_types)),
+                       available_instance_types=list(
+                           map(_text, instance_types)),
+                       available_disk_categories=list(
+                           map(_text, disk_categories)))
+
+    def _get_pagination(self, element):
+        page_number = int(findtext(element, 'PageNumber'))
+        total_count = int(findtext(element, 'TotalCount'))
+        page_size = int(findtext(element, 'PageSize'))
+        return Pagination(total=total_count, size=page_size,
+                          current=page_number)
+
+    def _request_multiple_pages(self, path, params, parse_func):
+        """
+        Request all resources by multiple pages.
+        :param path: the resource path
+        :type path: ``str``
+        :param params: the query parameters
+        :type params: ``dict``
+        :param parse_func: the function object to parse the response body
+        :param type: ``function``
+        :return: list of resource object, if not found any, return []
+        :rtype: ``list``
+        """
+        results = []
+        while True:
+            one_page = self.connection.request(path, params).object
+            resources = parse_func(one_page)
+            results += resources
+            pagination = self._get_pagination(one_page)
+            if pagination.next() is None:
+                break
+            params.update(pagination.to_dict())
+        return results
+
+
+def _parse_bool(value):
+    if isinstance(value, bool):
+        return value
+    if u(value).lower() == 'true':
+        return True
+    return False
+
+
+"""
+Define the extra dictionary for specific resources
+"""
+RESOURCE_EXTRA_ATTRIBUTES_MAP = {
```

Comment:
Can you please move this to the top after imports?

We tend to put global, module level constants at the top, immediately after imports.